### PR TITLE
initial support for generating knative builds

### PR DESCRIPTION
**Pull request by jstrachan**
using the `jenkins-x.yml` file to simplify knative build authoring

#1420

### CODE REVIEW

File: pkg/tests/files.go [hunk 1]
```diff
@@ -53,3 +54,28 @@ func AssertFilesExist(t *testing.T, expected bool, paths ...string) {
 		}
 	}
 }
+
+func AssertEqualFileText(t *testing.T, expectedFile string, actualFile string) error {
```
Comment (ccojocar) [inline pkg/tests/files.go]: I think, we can replace this file test helpers with `https://godoc.org/gotest.tools/fs`.

Comment (jstrachan) [reply pkg/tests/files.go]: awesome, I didn't know about those - will do thanks!

Comment (jstrachan) [reply pkg/tests/files.go]: lemme refactor the unit tests to reuse gotest.tools in a separate PR...

**Pull request merged**